### ISSUE TITLE
RegExp for modifyObstructiveCode replaces var names started with `top` with `self` when it shouldn't

### ISSUE THREAD

**Issue by indooorsman** [cypress-io/cypress#4794]
https://github.com/cypress-io/cypress/blob/b568e825452582ba21e9a26f88daa4edc6b1b587/packages/server/lib/util/security.js#L17

One of my file contains the below code:
```js
if (topTarget.window === topTarget) { ... }
```
in Cypress, It's replaced with 
```js
if (topTarget.window === selfTarget) { ... }
```
and it caused error.

Please fix it as soon as possible, many thanks <3

**Comment by jennifer-shehane** [comment]
Hey @indooorsman, this is intended behavior in Cypress in order to have sites tested in Cypress work correctly - as they are run from within an iframe - and some sites have framebusting code.

You can turn off this behavior by setting [`modifyObstructiveCode: false`](https://on.cypress.io/configuration#modifyObstructiveCode) in your configuration. 

I'd like to more fully understand the error you are getting and the unintended effect that Cypress seems to have caused in your application though. Can you post the error and explain the unexpected behavior more fully? Thanks.

**Comment by indooorsman** [comment]
@jennifer-shehane Thanks for your reply, the error is `selfTarget is not defined`.
The original code:
```
funcA(topTarget) {
  if (topTarget.window === topTarget) { ... }
}
```
In cypress it's replaced with:
```
funcA(topTarget) {
  if (topTarget.window === selfTarget) { ... }
}
```
Of course, `selfTarget` is not defined ...

**Comment by jennifer-shehane** [comment]
@indooorsman Thanks. Yeah, indeed - I agree this RegExp is too lenient in this situation and is definitely a problem. Thanks for clearing this up - we need to fix this.

**Comment by indooorsman** [comment]
@jennifer-shehane Great, thanks for your efficient actions 💯

**Comment by jennifer-shehane** [comment]
Released in [3.4.1](https://on.cypress.io/changelog#3-4-1)
